### ISSUE TITLE
Enum values for Microsoft.ServiceFabric.Services.Remoting.RemotingListener

### ISSUE THREAD

**Issue by MartinDemberger** [Azure/service-fabric-services-and-actors-dotnet#72]
In Microsoft.ServiceFabric.Services.Remoting.RemotingListener (3.0.456 / .Net-Standard) the values for enum RemotingListener and RemotingClient are wrong.
This can easily be fixed by assigning fixed values to the enum values.
As a work arround I have to use the int value and cast it to the enum.

I don't even understand why two of the three values aren't available in 3.0.

**Comment by amanbha** [comment]
@MartinDemberger For net core 2.0 only V2 remoting stack is supported that's why the enums have fewer values. If you create a full net framework project you will see all values as it still supports V1 and V2 remoting stack.

**Comment by amanbha** [comment]
@MartinDemberger I am closing this issue as its the expected behavior, please reopen the issue if you have further questions. Thanks!

**Comment by MartinDemberger** [comment]
@amanbha Sorry, but the issue isn't the question but the case that the enum-values have different values in Framework and Core.
It's currently not possible to mix Framework and Core Services.

To mix the Framework and Core Services I have to create a service-interface Project with .net Standard. In the Interface i set the assembly to V2. The Problem occures if the consumer of the Service is .net core. Now the consumer tries to use the Proxy with the enum value 1 which isn't specified in .net core.
If I set the Attribute to V1 the consumer use the enum value 0 which means V2 in .net core. This is also a fault. It should fail with a message which means "V1 isn't supported".

For me the only solution is to keep all services on .net Framework because one of my dependency isn't available in .net core now.

**Comment by amanbha** [comment]
@MartinDemberger Thanks for explaining it.
Adding @hrushib and @raunakpandya for full framework and dotnet core mix support.

**Comment by VikSab** [comment]
Hi

This issue is causing our build to break - we are in .net 4.6.1 - and want to use latest version of the Service Fabric - But the FabActUtil.exe is failing on build

We are unable to define a custom serialization attribute for our actors.

for some reason this is being resolved to using RemotingListener.V1Listener causing the FabActUtil to load a ActorRemotingProviderAttribute with 2 additional abstract methods that need to be implemented - but arent showing up when we try to create derived type of ActorRemotingProviderAttribute

[assembly: CustomActorRemotingProvider(RemotingListener = RemotingListener.V2Listener, RemotingClient = RemotingClient.V2Client)]
[assembly: CustomServiceRemotingProvider(RemotingListener = RemotingListener.V2Listener, RemotingClient = RemotingClient.V2Client)]

Here is the failure
------------------------------------------

System.TypeLoadException: Method 'CreateServiceRemotingListener' in type 'Dover.DataServices.ServiceFabric.Services.Remoting.Json.JsonActorRemotingProviderAttribute' from assembly 'ServiceFabric.JsonRemoting, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at System.ModuleHandle.ResolveType(RuntimeModule module, Int32 typeToken, IntPtr* typeInstArgs, Int32 typeInstCount, IntPtr* methodInstArgs, Int32 methodInstCount, ObjectHandleOnStack type)
   at System.ModuleHandle.ResolveTypeHandleInternal(RuntimeModule module, Int32 typeToken, RuntimeTypeHandle[] typeInstantiationContext, RuntimeTypeHandle[] methodInstantiationContext)
   at System.Reflection.RuntimeModule.ResolveType(Int32 metadataToken, Type[] genericTypeArguments, Type[] genericMethodArguments)
   at System.Reflection.CustomAttribute.FilterCustomAttributeRecord(CustomAttributeRecord caRecord, MetadataImport scope, Assembly& lastAptcaOkAssembly, RuntimeModule decoratedModule, MetadataToken decoratedToken, RuntimeType attributeFilterType, Boolean mustBeInheritable, Object[] attributes, IList derivedAttributes, RuntimeType& attributeType, IRuntimeMethodInfo& ctor, Boolean& ctorHasParameters, Boolean& isVarArg)
   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeModule decoratedModule, Int32 decoratedMetadataToken, Int32 pcaCount, RuntimeType attributeFilterType, Boolean mustBeInheritable, IList derivedAttributes, Boolean isDecoratedTargetSecurityTransparent)
   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeAssembly assembly, RuntimeType caType)
   at System.Attribute.GetCustomAttributes(Assembly element, Type attributeType, Boolean inherit)
   at System.Attribute.GetCustomAttribute(Assembly element, Type attributeType, Boolean inherit)
   at System.Reflection.CustomAttributeExtensions.GetCustomAttribute[T](Assembly element)
   at Microsoft.ServiceFabric.Actors.Remoting.ActorRemotingProviderAttribute.GetProvider(IEnumerable`1 types)
   at Microsoft.ServiceFabric.Actors.Runtime.ActorTypeInformation.Get(Type actorType)
   at FabActUtil.Tool.LoadActors(Assembly inputAssembly, IList`1 actorFilters, IList`1 actorTypes)
   at FabActUtil.Tool.LoadActors(ToolContext context)
   at FabActUtil.Tool.ProcessInput(ToolContext context)
   at FabActUtil.Tool.Run(ToolArguments arguments)
   at FabActUtil.Program.Main(String[] args)

-----------------------------------------------------------------------------------------------

**Comment by VikSab** [comment]
This issue seems to be related to using the new csproj format - even though the target framework is 4.6.1 - if the project is moved to new csproj format - its being detected as netstandard/netcoreapp

**Comment by amanbha** [comment]
@MartinDemberger  We will fix the enums in the next major release and Cumulative Update release as well.

@VikSab We will look into this issue with new csproj format, till then to unblock yourself, could you please use the old csproj format. The  VS templates are still generating older .csproj for full .net framework actor projects, so please don't change them until its supported for Actors project. I cannot confirm ETA for supporting the new .csproj at this point as their are other high work items we are working on, but will get to it as soon as possible. Will create a new issue to track this new .csproj format failure you are seeing.

**Comment by VikSab** [comment]
@amanbha - that is how we have gotten unblocked

**Comment by Scythen** [comment]
@VikSab That fixes the build issues but does not fix the runtime issues.
To fix runtime you also need the cast:
`[assembly: JsonServiceRemotingProvider(RemotingClient = (Microsoft.ServiceFabric.Services.Remoting.RemotingClient)1, RemotingListener = (Microsoft.ServiceFabric.Services.Remoting.RemotingListener)2)]`


The enums in the core and standard versions should have the same value as the framework versions, ie.
`    public enum RemotingClient
    {
        V2Client = 1
    }`
`    public enum RemotingListener
    {
        V2Listener = 2
    }`

**Comment by amanbha** [comment]
@Scythen the fix will be to explicitly add enum value.
+ @deepakjangid-MSFT , fyi.

public enum RemotingClient
    {
#if !DotNetCoreClr
        /// <summary>
        /// This is selected to create V1 Client. V1 is an old(soon to be deprecated) Remoting Stack.
        /// </summary>
        V1Client = 0,
#endif

        /// <summary>
        /// This is selected to create V2 Client. V2 is a new Remoting Stack.
        /// </summary>
        V2Client = 1,
    }

**Comment by amanbha** [comment]
Fix for this is in Microsoft.ServiceFabric.Services.Remoting nuget package version >=3.1.262
